server/routes/shop.test.js

```diff
@@ -0,0 +1,44 @@
+const request = require('supertest')
+
+const server = require('../server')
+
+const db = require('../db/shop')
+
+jest.mock('../db/shop')
+
+const mockProducts = [
```

Comment:
As I mentioned in server/db/shop.test.js - that is the place to test the DB data thoroughly.

Your test is still valid.

Since this is testing the server's shop route, it would be good to add an error handling test, like in news.test.js.